### PR TITLE
feat: use TxParser enum for TxMetadata['parser']

### PR DESCRIPTION
**Pull request by gomesalexandre**
re: https://github.com/shapeshift/unchained/pull/335#discussion_r833178486

This abstracts the `parser` property of `TxMetadata` into an enum for more type-safety, making sure we avoid casing issues when specifying a Tx parser.

### CODE REVIEW

File: node/packages/client/src/types.ts [hunk 1]
```diff
@@ -55,9 +55,16 @@ export enum TransferType {
   Receive = 'receive',
 }
 
+export enum TxParser {
+  Cosmos = 'cosmos',
+  Yearn = 'yearn',
+  UniV2 = 'uniV2',
+  ZRX = 'zrx',
+  THOR = 'thor',
```
Comment (kaladinlight) [inline node/packages/client/src/types.ts:63]: Referenced as `Thor` throughout files
```suggestion
  Thor = 'thor',
```

**Review by 0xApotheosis — APPROVED** [review]
Nice.

**Review by kaladinlight — APPROVED** [review]

**Review by elmutt — APPROVED** [review]

**Comment by elmutt** [comment]
Can we wait to merge this? I have an incoming PR that will overlap some of this stuff and merge conflicts suck. 

I can probably just include these changes in my PR too

**Comment by gomesalexandre** [comment]
@elmutt mergefixed

**Pull request merged**